engine/src/bug_stack.rs

```diff
@@ -0,0 +1,62 @@
+use crate::color::Color;
+use crate::piece::Piece;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct BugStack {
+    pub pieces: [Piece; 7],
+    pub size: u8,
+}
+impl Default for BugStack {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl BugStack {
+    pub fn new() -> Self {
+        Self {
+            pieces: [Piece::new(); 7],
+            size: 0,
+        }
+    }
+
+    pub fn len(&self) -> usize {
+        self.size as usize
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.size == 0
+    }
+
+    pub fn top_bug_color(&self) -> Option<Color> {
+        if self.is_empty() {
+            return None;
+        }
+        Some(self.pieces[self.size as usize].color())
+    }
+
+    pub fn push_piece(&mut self, piece: Piece) {
+        if self.size == 7 {
+            panic!("Trying to add an 8th bug to a BugStack")
```

Comment:
These only get called by the engine and not from any user input so panics are fine?